driver-examples/osgi/src/test/java/com/datastax/driver/osgi/MailboxServiceTest.java

```diff
@@ -0,0 +1,173 @@
+/*
+ *      Copyright (C) 2012-2015 DataStax Inc.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+package com.datastax.driver.osgi;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.GregorianCalendar;
+
+import javax.inject.Inject;
+import org.ops4j.pax.exam.Configuration;
+import org.ops4j.pax.exam.Option;
+import org.ops4j.pax.exam.options.CompositeOption;
+import org.ops4j.pax.exam.options.MavenArtifactProvisionOption;
+import org.ops4j.pax.exam.spi.reactors.ExamReactorStrategy;
+import org.ops4j.pax.exam.spi.reactors.PerMethod;
+import org.ops4j.pax.exam.testng.listener.PaxExam;
+import org.testng.annotations.Listeners;
+import org.testng.annotations.Test;
+
+import static org.ops4j.pax.exam.CoreOptions.bundle;
+import static org.ops4j.pax.exam.CoreOptions.mavenBundle;
+import static org.ops4j.pax.exam.CoreOptions.options;
+import static org.ops4j.pax.exam.CoreOptions.systemPackages;
+import static org.ops4j.pax.exam.CoreOptions.systemProperty;
+import static org.testng.Assert.assertEquals;
+
+import com.datastax.driver.core.CCMBridge;
+import com.datastax.driver.osgi.api.MailboxException;
+import com.datastax.driver.osgi.api.MailboxMessage;
+import com.datastax.driver.osgi.api.MailboxService;
+
+import static com.datastax.driver.osgi.VersionProvider.projectVersion;
+
+@ExamReactorStrategy(PerMethod.class)
+@Listeners({CCMBridgeListener.class, PaxExam.class})
+@Test(groups="short")
+public class MailboxServiceTest {
+    @Inject MailboxService service;
+
+    private MavenArtifactProvisionOption driverBundle = mavenBundle("com.datastax.cassandra", "cassandra-driver-core", projectVersion());
+
+    private MavenArtifactProvisionOption guavaBundle = mavenBundle("com.google.guava", "guava", "14.0.1");
+
+    private MavenArtifactProvisionOption nettyBundle = mavenBundle("io.netty", "netty", "3.9.0.Final");
+
+    private CompositeOption defaultOptions = new CompositeOption() {
+
+        @Override public Option[] getOptions() {
+            return options(
+                systemProperty("cassandra.contactpoints").value(CCMBridge.IP_PREFIX + 1),
+                bundle("reference:file:target/classes"),
+                mavenBundle("com.codahale.metrics", "metrics-core", "3.0.2"),
```

Comment:
One weakness here is that dependencies and their versions are hardcoded here, creates another place we have to manage versions which is less than ideal.